### DISCUSSION TITLE
[Feature] ability for remote checkmark to represent an upload not just to immich, but to the cloud (and or functionality to back up to cloud)

### DISCUSSION THREAD

**Discussion by terencechow** [immich-app/immich#13451] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, when you look through the app you can see what is stored remotely to your server. However this is just my server. I actually upload this to AWS because I don't trust myself to safely store all this data without losing it. (This is stored at an archive storage class for cheapest price).

I'd like to be able to look at the immich app, whether through my phone or browser, and understand the remote checkmark actually means i've uploaded this to the cloud because right now.

Basically, when my phone hits its storage limit, I need to cross reference files on my immich server with the ones stored from s3 and then go on my phone and delete the ones from my phone. It would be much easier if i could look at the check mark and know that it already exists in s3 and feel comfortable deleting it.

I think this could possibly be done as part of administration tab. Essentially, the adminsitration tab allows some way of putting in some access keys etc and then when enabled, will upload data from the immich server to s3 periodically, with a storage class determined by this administration tab. I'm not sure how the remote flag is done (ie is it just checking file existence or is it some database), but if its a database, then something in the database can mark a flag ''saved to cloud', etc. (In the absence of a built in feature, maybe the remote checkmark flag can be saved to db only after some user provided hook, ie a script through the docker container? not sure about this, i think the fully built feature is safer from a security perspective) 

I think its  common to back up to the cloud for a lot of home servers, so having this extra functionality would be beneficial as it give the home server extra confidence that this file is backed up

### Platform

- [x] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Everyone's backup system is sufficiently different that I don't think it's feasible for Immich to build in explicit support for stuff like this.
What I'd recommend as an alternative is to give yourself an easy way to trigger a server backup on-demand. Then you can just run that every time before you start deleting files.

**Reply by terencechow** [reply]
Maybe my example was too specific for my use case, ie s3, but I was thinking about it in a more general way. Specifically if the admin panel had the relevant place to put access keys and secrets, i'm not sure why it cannot extend to any backend that supports the s3 api. For reference i'm thinking about having the docker image for immich-server, install [rclone](https://rclone.org/) (which supports over 70 cloud providers who have similar apis as aws' s3). It might be as easy as spawning a new rclone process at some periodic cadence, that simply runs `rclone copy src dest`. Does that seem generic enough to be useful?

How is the remote checkmark determined? I'm having trouble finding anything in the database that might be what it refers to. Is it just an fstat for the file from the immich server?

**Reply by bo0tzz** [reply]
> rclone copy src dest

A copy is not a backup.

> How is the remote checkmark determined?

It's displayed if the app has a record in its internal database showing that the asset exists in the server's database.
